test/unit/vm/disposable_test.dart

```diff
@@ -28,6 +28,32 @@ void main() {
       thing = new DisposableThing();
     });
 
+    group('disposalTreeSize', () {
+      test('should count all managed objects', () {
+        var controller = new StreamController();
+        var subscription = controller.stream.listen((_) {});
+        thing.manageStreamController(controller);
+        thing.manageStreamSubscription(subscription);
+        thing.manageDisposable(new DisposableThing());
+        thing.manageCompleter(new Completer());
+        thing.getManagedTimer(new Duration(days: 1), () {});
+        thing
+            .getManagedDelayedFuture(new Duration(days: 1), () {})
+            .catchError((_) {}); // Because we dispose prematurely.
+        thing.getManagedPeriodicTimer(new Duration(days: 1), (_) {});
+        expect(thing.disposalTreeSize, 8);
+        thing.dispose();
+      });
+
+      test('should count nested objects', () {
+        var nestedThing = new DisposableThing();
+        nestedThing.manageDisposable(new DisposableThing());
+        thing.manageDisposable(nestedThing);
+        expect(thing.disposalTreeSize, 3);
+        thing.dispose();
```

Comment:
Here too it might be worth confirming the disposalTreeSize matches some expectation post dispose().